openapi.yml

```diff
@@ -0,0 +1,576 @@
+    description: Existing secure message functionality
+  - name: New
+    description: New to secure message v2
+paths:
+  /messages:
+    post:
+      tags:
+        - Messages
+        - Existing
+      summary: Post a new message to a thread
+      description: Create an additional message linked to an already created thread
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                thread_id:
+                  type: string
+                  format: uuid
+                  example: 91e0bb4f-ffad-404c-af1b-01c87037119c
+                body:
+                  type: string
+                  example: I would like some help with this survey
+                is_from_internal:
+                  type: boolean
+                  example: true
+                sent_by:
+                  type: string
+                  format: uuid
+                  example: 6934f0c7-b595-435b-aa8f-291d6b6070d3
+      responses:
+        201:
+          description: Message created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Message'
+        400:
+          description: Payload was incorrect
+        405:
+          description: Method not allowed
+  /messages/{message_id}:
+    patch:
+      tags:
+        - Messages
+        - New
+      summary: Patch a message by the message_id
+      description: Update a message's data by the `message_id`
+      parameters:
+        - name: message_id
+          in: path
+          description: UUID of the message
+          required: true
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        required: false
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                body:
+                  type: string
+                  example: I cannot access this survey
+                sent_by:
+                  type: string
+                  format: uuid
+                  example: 6934f0c7-b595-435b-aa8f-291d6b6070d3
+      responses:
+        200:
+          description: Message patched
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Message'
+        400:
+          description: Client Error
+        404:
+          description: Message not found
+        405:
+          description: Method not allowed
+  /threads:
+    get:
+      tags:
+        - Threads
+        - Existing
+      summary: Get threads with certain parameters
+      description: Get a filtered list of threads based on a number of supplied query parameters
+      parameters:
+        - name: ru_ref
+          in: query
+          description: A string that refers to a specific business
+          schema:
+            type: string
+            example: 41104403458
```

Comment:
There is an existing parameter `business_id` which takes the UUID of the business party, but I can't see an existing filter for `ru_ref`.

Reply:
We no longer hold the business_id in the database and in rOps the messages can only be filtered by: `category`(required), `survey_id`(required), or `ru_ref`. In frontstage they cannot be filtered